res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -24,14 +25,126 @@ var primeHotcueColors = new ColorMapper({
   0xFF00FF: 19, // full magenta
 });
 
+// common LED values for non-RGB lights
+Prime4.colorCodes = {
+  'off': 0x00,
+  'dim': 0x01,
+  'on': 0x7F
+}
+
+
+
+Prime4.activeButtons = {};
+
+Prime4.unshiftedButtons = {
+  censor = function (channel, control, value, status, group) {
+    // code for reverse + slip mode
+  }
+};
+
+Prime4.shiftedButtons = {
+  censor = function (channel, control, value, status, group) {
+    // code for reverse mode
+  }
+};
+
 Prime4.init = function (id, debugging) {
-	// initializing script
+	Prime4.activeButtons = Prime4.unshiftedButtons; // initialize script with shift button unpressed
+  Prime4.initDeck('[Channel1]')
+  Prime4.initDeck('[Channel2]')
 }
 
-Prime4.controllerTest = function  (channel, control, value, status, group) {
-	// controller mapping script
+// code for the shift buttons
+Prime4.shiftButton = function (channel, control, value, status, group) {
+  if (value === 127) {
+    engine.connectControl(group, key, true);
+    Prime4.activeButtons = Prime4.shiftedButtons;
+    engine.connectControl(group, key);
+  } else {
+    engine.connectControl(group, key, true);
+    Prime4.activeButtons = Prime4.unshiftedButtons;
+    engine.connectControl(group, key);
+  }
 }
 
 Prime4.shutdown = function () {
 	// shutdown script
 }
+
+
+
+
+
+
+
+// allowing the Prime 4 to control all four decks as intended
+
+Prime4.deck = {
+  '[Channel1]': '[Channel1]',
+  '[Channel2]': '[Channel2]'
+}
+
+Prime4.buttons = {
+  '[Channel1]': {
+    'deckToggle': // deck 1/3 LED MIDI code
+  }
+  '[Channel2]': {
+    'deckToggle': // deck 2/4 LED MIDI code
+  }
+}
+Prime4.buttons['[Channel3]'] = Prime4.buttons['[Channel1]']
+Prime4.buttons['[Channel4]'] = Prime4.buttons['[Channel2]']
+
+Prime4.channelRegEx = /\[Channel(\d+)\]/
+Prime4.deckToggleButton = function (channel, control, value, status, group) {
+  if (value) {
+    var deckNumber = parseInt(
+      Prime4.channelRegEx.exec(
+        Prime4.deck[group]
+      )[1]
+    )
```

Comment:
we already provide a utility function for this: `script.deckFromGroup(group)`. See: https://github.com/mixxxdj/mixxx/wiki/midi%20scripting#Helper-functions